platform_secrets/service.py

```diff
@@ -51,48 +120,58 @@ def create(cls) -> NamespaceForbiddenError:
 @dataclass(frozen=True)
 class Secret:
     key: str
+    org_name: str
     project_name: str
     value: str = field(repr=False, default="")
-    org_name: Optional[str] = None
+
+    @property
+    def namespace_name(self) -> str:
+        return generate_namespace_name(self.org_name, self.project_name)
 
 
 class Service:
-    def __init__(self, kube_client: KubeClient) -> None:
-        self._kube_client = kube_client
+    def __init__(self, kube_api: KubeApi) -> None:
+        self._kube_api = kube_api
 
     def _get_kube_secret_name(self, project_name: str, org_name: Optional[str]) -> str:
         path = project_name
         if org_name:
+            org_name = NO_ORG_NORMALIZED if org_name == NO_ORG else org_name
             path = f"{org_name}/{path}"
         return f"project--{path.replace('/', '--')}--secrets"
 
     def _get_project_name_from_secret_name(
-        self, secret_name: str, org_name: Optional[str]
+        self, secret_name: str, org_name: str
```

Comment:
should org_name remain optional?

Reply:
@zubenkoivan I've actually made it required everywhere on the service layer, and handling the presence of it on the API layer.